### PR TITLE
Makes service method API invocations over HTTP by default.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Makes service method API invocations over HTTP by default.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #438 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: sdk/src/main/java/io/dapr/client/DaprClientBuilder.java [hunk 1]
```diff
@@ -97,11 +103,30 @@ public DaprClientBuilder withStateSerializer(DaprObjectSerializer stateSerialize
...
+   */
+  private DaprClient buildDaprClient(DaprApiProtocol protocol) {
+    if (protocol == null) {
+      throw new IllegalArgumentException("Protocol is required.");
+    }
+
+    switch (protocol) {
+      case GRPC: return buildDaprClientGrpc();
+      case HTTP: return buildDaprClientHttp();
+      default: throw new IllegalArgumentException("Unsupported protocol: " + protocol.name());
```
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/client/DaprClientBuilder.java:128]: Why is this exception different and worded differently from the exception in `ActorProxyBuilder`?
There it was:
```java
switch (this.apiProtocol) {
      return new DaprGrpcClient(DaprGrpc.newFutureStub(this.channel));	      case GRPC: return new DaprGrpcClient(DaprGrpc.newFutureStub(this.channel));
      case HTTP: return new DaprHttpClient(daprHttpBuilder.build());
      default: throw new IllegalStateException("Invalid Dapr API protocol.");
}
```

File: sdk/src/main/java/io/dapr/config/Properties.java [hunk 2]
```diff
@@ -24,14 +26,19 @@
...
-   * Dapr's default GRPC port.
+   * Dapr's default use of gRPC or HTTP.
+   */
+  private static final DaprApiProtocol DEFAULT_API_PROTOCOL = DaprApiProtocol.GRPC;
+
+  /**
+   * Dapr's default use of gRPC or HTTP for Dapr's method invocation APIs.
    */
-  private static final Boolean DEFAULT_GRPC_ENABLED = true;
+  private static final DaprApiProtocol DEFAULT_API_METHODINVOCATION_PROTOCOL = DaprApiProtocol.HTTP;
```
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/config/Properties.java:41]:
```suggestion
  private static final DaprApiProtocol DEFAULT_API_METHOD_INVOCATION_PROTOCOL = DaprApiProtocol.HTTP;

```

File: sdk/src/main/java/io/dapr/config/Properties.java [hunk 3]
```diff
@@ -68,12 +75,22 @@
...
+
+  /**
+   * Determines if Dapr client should use gRPC or HTTP for Dapr's service method invocation APIs.
    */
-  public static final Property<Boolean> USE_GRPC = new BooleanProperty(
-      "dapr.grpc.enabled",
-      "DAPR_GRPC_ENABLED",
-      DEFAULT_GRPC_ENABLED);
+  public static final Property<DaprApiProtocol> API_METHODINVOCATION_PROTOCOL = new GenericProperty<>(
+      "dapr.api.methodInvocation.protocol",
```
Comment (mukundansundar) [inline sdk/src/main/java/io/dapr/config/Properties.java:90]:
```suggestion
      "dapr.api.method.invocation.protocol",

```

**Review by mukundansundar — COMMENTED** [review]
I can understand why you left it as methodInvocation in the propery and METHODINVOCATION in the env var, but I think method.invocation and METHOD_INVOCATION looks better and follows convention.

File: sdk/src/main/java/io/dapr/config/Properties.java (hunk 3 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/config/Properties.java:90]: This is wrong because method is not a child of api.

File: sdk/src/main/java/io/dapr/config/Properties.java (hunk 2 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/config/Properties.java:41]: I can make this change simply because it is internal and reads as per JAVA convention.

File: sdk/src/main/java/io/dapr/client/DaprClientBuilder.java (hunk 1 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/client/DaprClientBuilder.java:128]: OK. Made them the same. IllegalState because it is the state of the builder that does not depend on a param.

File: sdk/src/main/java/io/dapr/config/Properties.java (hunk 2 above)
Comment (artursouza) [reply sdk/src/main/java/io/dapr/config/Properties.java:41]: I also renamed another one and renamed the env variables.

File: sdk-actors/src/main/java/io/dapr/actors/runtime/ActorRuntime.java [hunk 4]
```diff
@@ -320,10 +321,6 @@ private static DaprClient buildDaprClient(ManagedChannel channel) {
    * @return GRPC managed channel or null.
    */
   private static ManagedChannel buildManagedChannel() {
-    if (!Properties.USE_GRPC.get()) {
```
Comment (mukundansundar) [inline sdk-actors/src/main/java/io/dapr/actors/runtime/ActorRuntime.java:323]: why is this removed?

Comment (artursouza) [reply sdk-actors/src/main/java/io/dapr/actors/runtime/ActorRuntime.java:323]: Actually, this is needed in the case actors communicate over HTTP. Good catch.

**Review by mukundansundar — APPROVED** [review]
